static/scripts/rewards/render-transaction/render-transaction.ts

```diff
@@ -1,81 +1,76 @@
 import { ERC20Permit, Permit, TokenType } from "@ubiquibot/permit-generation/types";
 import { networkExplorers } from "@ubiquity-dao/rpc-handler";
 import { app } from "../app-state";
-import { buttonController, getMakeClaimButton, viewClaimButton } from "../toaster";
+import { buttonControllers, getMakeClaimButton, getViewClaimButton } from "../toaster";
 import { checkRenderInvalidatePermitAdminControl, claimErc20PermitHandlerWrapper, fetchTreasury } from "../web3/erc20-permit";
 import { claimErc721PermitHandler } from "../web3/erc721-permit";
 import { verifyCurrentNetwork } from "../web3/verify-current-network";
 import { insertErc20PermitTableData, insertErc721PermitTableData } from "./insert-table-data";
 import { renderEnsName } from "./render-ens-name";
 import { renderNftSymbol, renderTokenSymbol } from "./render-token-symbol";
 
-const carousel = document.getElementById("carousel") as Element;
-const table = document.querySelector(`table`) as HTMLTableElement;
-type Success = boolean;
-
-export async function renderTransaction(): Promise<Success> {
-  if (app.claims && app.claims.length > 1) {
-    carousel.className = "ready";
-    const rewardsCount = document.getElementById("rewardsCount") as Element;
-    rewardsCount.innerHTML = `${app.rewardIndex + 1}/${app.claims.length} reward`;
-  }
-
-  if (!app.reward) {
-    buttonController.hideAll();
+export async function renderTransaction(claim: Permit, table: Element): Promise<boolean> {
+  if (!claim) {
+    buttonControllers[claim.nonce].hideAll();
     console.log("No reward found");
     return false;
   }
 
-  verifyCurrentNetwork(app.reward.networkId).catch(console.error);
+  if (!table) {
+    throw new Error("Missing transaction table");
+  }
+
+  verifyCurrentNetwork(claim.networkId).catch(console.error);
```

Comment:
You call `renderTransaction()` while creating each new table so this will fire multiple network change prompts from MM if multiple networks exist and ultimately it'll prevent at least claim from being claimed